scripts/survived.py

```
@@ -0,0 +1,79 @@
+        elif line.startswith("+") and not line.startswith("+++") and cur:
+            body = line[1:]
+            if len(body.strip()) >= 25 and not TRIVIAL.match(body):
+                by_file[cur].append(body.strip())
+    return {f: v for f, v in by_file.items() if v}
+
+
+def survives(ref, path, lines):
+    blob = git("show", "%s:%s" % (ref, path))
+    if not blob:
+        return 0, len(lines)          # file gone entirely
+    kept = sum(1 for l in lines if l in blob)
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)